eth/catalyst/api.go

```diff
@@ -207,21 +207,21 @@ func (api *ConsensusAPI) verifyPayloadAttributes(attr *engine.PayloadAttributes)
 	c := api.eth.BlockChain().Config()
 
 	// Verify withdrawals attribute for Shanghai.
-	if err := checkAttribute(c.IsShanghai, attr.Withdrawals != nil, attr.Timestamp); err != nil {
+	if err := checkAttribute(c.IsShanghai, attr.Withdrawals != nil, c.LondonBlock, attr.Timestamp); err != nil {
 		return fmt.Errorf("invalid withdrawals: %w", err)
 	}
 	// Verify beacon root attribute for Cancun.
-	if err := checkAttribute(c.IsCancun, attr.BeaconRoot != nil, attr.Timestamp); err != nil {
+	if err := checkAttribute(c.IsCancun, attr.BeaconRoot != nil, c.LondonBlock, attr.Timestamp); err != nil {
 		return fmt.Errorf("invalid parent beacon block root: %w", err)
 	}
 	return nil
 }
 
-func checkAttribute(active func(*big.Int, uint64) bool, exists bool, time uint64) error {
-	if active(common.Big0, time) && !exists {
+func checkAttribute(active func(*big.Int, uint64) bool, exists bool, block *big.Int, time uint64) error {
```

Comment:
The london block is used to align with line https://github.com/ethereum/go-ethereum/pull/28135/files#diff-96840e4882901b94cef9f51a992a44e64df659335ff13dd011be8c1ffa097135R179

Although I don't think london block is correct here. The block number to generate is the one to pass, IIUC.

Reply:
The original code passed the weird london block because in merge mode we *know* that we already passed london. So instead of complicating everything with passing a block number, it was a hacky way to hard code "always london".